slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -220,12 +223,174 @@ trait PostgresProfile extends JdbcProfile {
+    override val localTimeType = new LocalTimeJdbcType
+    override val offsetTimeType = new OffsetTimeJdbcType
+    override val instantType = new InstantJdbcType
+    override val localDateTimeType = new LocalDateTimeJdbcType
 
     class ByteArrayJdbcType extends super.ByteArrayJdbcType {
       override val sqlType = java.sql.Types.BINARY
       override def sqlTypeName(sym: Option[FieldSymbol]) = "BYTEA"
     }
 
+    trait PostgreTimeJdbcType [T] {
+
+      implicit val min : T
+      implicit val max : T
+      implicit val serialize : (T => String)
+      implicit val parse : (String => T)
+
+      @inline
+      private[this] val negativeInfinite = "-infinity"
+      @inline
+      private[this] val positiveInfinite = "infinity"
+
+      protected def serializeTime(time : T): String = {
+        time match {
+          case null => null
+          case `min` => negativeInfinite
+          case `max` => positiveInfinite
+          case _ => serialize(time)
+        }
+      }
+      protected def parseTime(time : String): T = {
+        time match {
+          case null => null.asInstanceOf[T]
+          case `negativeInfinite` => max
+          case `positiveInfinite` => min
+          case _ => parse(time)
+        }
+      }
+    }
+
+    class LocalDateJdbcType extends super.LocalDateJdbcType with PostgreTimeJdbcType[LocalDate] {
+
+      private[this] val formatter = DateTimeFormatter.ISO_LOCAL_DATE
+
+      implicit val min : LocalDate = LocalDate.MIN
+      implicit val max : LocalDate = LocalDate.MAX
+      implicit val serialize : (LocalDate => String) =  _.format(formatter)
+      implicit val parse : (String => LocalDate) = LocalDate.parse(_, formatter)
+
+      override val sqlType = java.sql.Types.OTHER
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "DATE"
+      override def getValue(r: ResultSet, idx: Int): LocalDate = parseTime(r.getString(idx))
+      override def setValue(v: LocalDate, p: PreparedStatement, idx: Int) = {
+        p.setObject(idx, serializeTime(v), sqlType)
+      }
+      override def updateValue(v: LocalDate, r: ResultSet, idx: Int) = {
+        r.updateObject(idx, serializeTime(v), sqlType)
+      }
+      override val hasLiteralForm : Boolean = false
+    }
+
+    class LocalTimeJdbcType extends super.LocalTimeJdbcType with PostgreTimeJdbcType[LocalTime] {
+
+      private[this] val formatter : DateTimeFormatter = DateTimeFormatter.ISO_LOCAL_TIME
+
+      implicit val min : LocalTime = LocalTime.MIN
+      implicit val max : LocalTime = LocalTime.MAX
+      implicit val serialize : (LocalTime => String) =  _.format(formatter)
+      implicit val parse : (String => LocalTime) = LocalTime.parse(_, formatter)
+
+      override val sqlType = java.sql.Types.OTHER
+      override def sqlTypeName(sym: Option[FieldSymbol]) = "TIME"
+      override def setValue(v: LocalTime, p: PreparedStatement, idx: Int) = {
+        p.setObject(idx, serialize(v), sqlType)
```

Comment:
In slick-pg, when calling `p.setObject`, I always wrapped the value in a `PGobject`.
But I just tested it on local, serialized string is also OK, and better since it doesn't depend on a driver specific class. :-)